Classes/Emogrifier.php

```diff
@@ -430,6 +431,28 @@ protected function process(\DOMDocument $xmlDocument)
     }
 
     /**
+     * Searches for all nodes with a style attribute, transforms the CSS found
+     * to HTML attributes and adds those attributes to each node.
+     *
+     * @param \DOMXPath $xPath
+     *
+     * @return void
+     */
+    private function mapAllInlineStylesToHtmlAttributes(\DOMXPath $xPath)
+    {
+        $nodesWithStyleAttributes = $xPath->query('//*[@style]');
+        if ($nodesWithStyleAttributes == false) {
```

Comment:
Generally, please always use strict comparisons: `=== false`

Reply:
The check for `false` (the error case) should indeed be enough. If the node list is empty (which is a perfectly valid case), the `foreach` will just be a no-oop.

Actually, I think we should throw a `\RuntimeException` if the result is false because then either the expression is invalid (which would be a bug in Emogrifier) or the `$xPath` would be invalid (we which should catch earlier). Please also add the corresponding `@throws` annotation.

Reply:
I love how refactoring brings buried subtle bugs and not-so-clean programming to the surface! :heart:

Reply:
Throw the exception in debug mode only?

Reply:
Yes, that makes sense.